Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.cpp

```diff
@@ -274,11 +258,18 @@ void LegacyRenderSVGShape::fillStrokeMarkers(PaintInfo& childPaintInfo)
         case PaintType::Stroke:
             strokeShape(style(), childPaintInfo.context());
             break;
-        case PaintType::Markers:
-            if (!m_markerPositions.isEmpty())
-                drawMarkers(childPaintInfo);
+        case PaintType::Markers: {
+            switch (m_shapeType) {
+            case ShapeType::Line:
+            case ShapeType::Path:
+                downcast<LegacyRenderSVGPath>(*this).drawMarkers(childPaintInfo);
```

Comment:
Why do we have to cast here from `LegacyRenderSVGShape` to `LegacyRenderSVGPath`? Can't we use kind of polymorphism to avoid this casting?

Reply:
I added this not to do much virtual function calls when we already know what class we need to dispatch. But I'm ok to have `drawMarkers` virtual function here since I know this part is not getting perf concerns from the benchmark right now. Changed.